### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by anatolii-zarytskyi**

### CODE REVIEW

**Review by iedenysenko — APPROVED** [review]
good :)

File: src/main/java/practice/CandidateValidator.java [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+import model.Candidate;
 
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final int RESIDENCE_YEAR = 10;
+    private static final String NATIONALITY = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] periodInUkraine = candidate.getPeriodsInUkr().split("-");
```
Comment (KseniiaMakarova) [inline src/main/java/practice/CandidateValidator.java:13]: let's make this a constant as well

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -14,7 +17,12 @@
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(num -> Arrays.stream(num.split(",")))
+                .mapToInt(Integer::parseInt)
+                .filter(n -> n % 2 == 0)
```
Comment (KseniiaMakarova) [inline src/main/java/practice/StreamPractice.java:23]:
```suggestion
                .flatMap(number -> Arrays.stream(number.split(",")))
                .mapToInt(Integer::parseInt)
                .filter(number -> number % 2 == 0)
```

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -35,7 +47,11 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
      */
     public List<People> selectMenByAge(List<People> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(people -> people.getAge() >= fromAge
+                        && people.getAge() <= toAge
+                        && people.getSex() == People.Sex.MAN)
```
Comment (KseniiaMakarova) [inline src/main/java/practice/StreamPractice.java:53]: let's create a Predicate instance as the checklist suggests

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -50,7 +66,13 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      */
     public List<People> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<People> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(people -> people.getSex() == People.Sex.MAN
+                        && people.getAge() >= fromAge && people.getAge() <= maleToAge
+                        || (people.getSex() == People.Sex.WOMEN)
+                        && people.getAge() >= fromAge && people.getAge() <= femaleToAge)
```
Comment (KseniiaMakarova) [inline src/main/java/practice/StreamPractice.java:73]: same here
also, `people.getAge() >= fromAge` is checked in both cases
can we optimize the code?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/practice/CandidateValidator.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/practice/CandidateValidator.java]: regex

File: src/main/java/practice/StreamPractice.java [hunk 5]
```diff
@@ -50,7 +69,14 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      */
     public List<People> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<People> peopleList) {
-        return Collections.emptyList();
+        Predicate<People> peoplePredicate = people -> people.getAge() >= fromAge
```
Comment (KseniiaMakarova) [inline src/main/java/practice/StreamPractice.java:72]: let's better have different names for different predicates

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**